docroot/profiles/custom/sitenow/sitenow.install

```diff
@@ -1950,3 +1951,116 @@ function sitenow_update_9054() {
+       * @var \Drupal\field\Entity\FieldConfig $definition
+       */
+      foreach ($fields as $field_name => $definition) {
+        if (in_array($definition->getType(), [
+          'text_long',
+          'text_with_summary',
+        ])) {
+          $field_check[$field_name] = $definition;
+        }
+      }
+
+      // Only need to run the following if we found fields to check against.
+      if (!empty($field_check)) {
+
+        // Retrieve all entities of this bundle type;.
+        // Media is listed as 'bundle,' but other entity types
+        // are listed as 'type'.
+        $type = $entity_type == 'media' ? 'bundle' : 'type';
+        $entities = \Drupal::entityTypeManager()->getStorage($entity_type)
+          ->loadByProperties([$type => $bundle]);
+
+        // Loop through the loaded entities.
+        foreach ($entities as $entity) {
+          $needs_save = FALSE;
+
+          // Loop through the fields we defined before.
+          foreach ($field_check as $name => $definition) {
+            // If the field has a value (or values).
+            if (!$entity->get($name)->isEmpty()) {
+              $updated_values = [];
+              $field_values = $entity->{$name}->getValue();
+
+              foreach ($field_values as $field_value) {
+
+                // @todo Any special handling for text_with_summary fields?
+                $document = Html::load($field_value['value']);
+
+                // Get all <drupal-media> instances.
+                $dm_instances = $document->getElementsByTagName('drupal-media');
+
+                // Built with this as a template.
+                // https://github.com/uiowa/uiowa/pull/3048/commits/5b95c9912aaa2b235cf06787a8c673a0f2cb17d7
+                $i = $dm_instances->length - 1;
+
+                while ($i >= 0) {
+                  // The current inline image element.
+                  $old_dm_instance = $dm_instances->item($i);
+                  $new_dm_instance = $old_dm_instance;
+
+                  $view_mode = $old_dm_instance->getAttribute('data-view-mode');
+                  $is_relevant_view_mode = in_array($view_mode, $media_view_mode_map_keys);
+                  if ($is_relevant_view_mode) {
+                    $new_dm_instance->setAttribute('data-view-mode', $media_view_mode_map[$view_mode]);
+
+                    $old_dm_instance->parentNode->replaceChild($new_dm_instance, $old_dm_instance);
+                  }
+
+                  $i--;
+                }
+
+                $updated_values[] = [
+                  'value' => Html::serialize($document),
+                  'format' => $field_value['format'],
+                ];
+              }
+
+              if (count($updated_values) === 1) {
+                $updated_values = $updated_values[0];
+              }
+
+              $entity->{$name} = $updated_values;
+              $needs_save = TRUE;
+            }
+          }
+          if ($needs_save) {
+            $entity->save();
+          }
```

Comment:
Do we want to set a revision/revision message for this?

On a similar note, do we want/need to account for revisions in this update, in case someone reverts to a previous revision after the update hook runs, and ends up with the old view mode?
```suggestion
          if ($needs_save) {
            $entity->setNewRevision(TRUE);
            $entity->revision_log = 'Updated entity embed view modes.';
            $entity->save();
          }
```

Reply:
Looking at the update hook in a little closer detail, I'm wondering if this should be re-written to access the database tables directly to pull the field value and update it. My impression is that the entity load is going to be a much heavier process. We might need to consider batching this anyway for sites that have a very large number of nodes/blocks, but using the direct database method might make that a moot point. That would also allow us to not have to load an entity to discover whether it contains a value that we need to work on, as that could just be reflected in the query. Finally, we could use logic like https://git.drupalcode.org/project/drupal/-/blob/c540e9ae4917193f75bdc539ce66858fa3f1e49e/core/modules/layout_builder/layout_builder.install#L106-113 to programmatically determine whether revisions need to be considered. 

And to answer that question, yes, we need to account for revisions. The alternative is to remove old revisions, which is not desirable. We essentially need to set the update up to make it as though the old view modes never existed.

Reply:
> Finally, we could use logic like https://git.drupalcode.org/project/drupal/-/blob/c540e9ae4917193f75bdc539ce66858fa3f1e49e/core/modules/layout_builder/layout_builder.install#L106-113 to programmatically determine whether revisions need to be considered.

Also just remembered the tempstore will have to be considered (which is part of update hook linked above, in the `key_value_expire` section).